landoapi/api/transplants.py

```diff
@@ -388,18 +389,21 @@ def post(phab: PhabricatorClient, data: dict):
 
         # Trigger a local transplant
         job = LandingJob(
-            status=LandingJobStatus.SUBMITTED,
             requester_email=ldap_username,
             repository_name=landing_repo.short_name,
             repository_url=landing_repo.url,
-            revision_to_diff_id=revision_to_diff_id,
-            revision_order=revision_order,
         )
 
         db.session.add(job)
     db.session.commit()
```

Comment:
Yes, the job ID is set on db commit, and job ID needs to exist before related fields can be added to the association table.

Reply:
Ah, I knew I had done this for a reason. So if we did not care about the order of revisions, then this would work fine without the commit. The problem is that if we do not commit the db session, then `job.sort_revisions` (which is called in `job.add_revisions`) will not work as expected, as it performs update statements based on a lookup by job and revision IDs. Helped to add the test as I am now able to reproduce this in the test. I'll see if I can simplify the commits so that they are not scattered all over the place.